src/esp32/entry.rs

```diff
@@ -24,81 +24,76 @@ use esp_idf_svc::http::client::{Configuration as HttpConfiguration, EspHttpConne
 pub fn serve_web(
     app_config: AppClientConfig,
     tls_server_config: Esp32TlsServerConfig,
-    robot: Option<LocalRobot>,
+    initializer: Initializer,
     _ip: Ipv4Addr,
     webrtc_certificate: WebRtcCertificate,
 ) {
-    let (mut srv, robot) = {
-        let mut client_connector = Esp32Tls::new_client();
-        let exec = Esp32Executor::new();
-        let mdns = Esp32Mdns::new("".to_string()).unwrap();
+    let mut client_connector = Esp32Tls::new_client();
+    let exec = Esp32Executor::new();
+    let mdns = Esp32Mdns::new("".to_string()).unwrap();
 
-        let cfg_response = {
-            let cloned_exec = exec.clone();
-            let conn = client_connector.open_ssl_context(None).unwrap();
-            let conn = Esp32Stream::TLSStream(Box::new(conn));
-            let grpc_client =
-                Box::new(GrpcClient::new(conn, cloned_exec, "https://app.viam.com:443").unwrap());
-
-            let builder = AppClientBuilder::new(grpc_client, app_config.clone());
-
-            let mut client = builder.build().unwrap();
-
-            client.get_config().unwrap()
-        };
+    let cfg_response = {
+        let cloned_exec = exec.clone();
+        let conn = client_connector.open_ssl_context(None).unwrap();
+        let conn = Esp32Stream::TLSStream(Box::new(conn));
+        let grpc_client =
+            Box::new(GrpcClient::new(conn, cloned_exec, "https://app.viam.com:443").unwrap());
 
-        let robot = match robot {
-            Some(r) => Arc::new(Mutex::new(r)),
-            None => {
-                log::info!("building robot from config");
-                let r = LocalRobot::new_from_config_response(&cfg_response).unwrap();
-                Arc::new(Mutex::new(r))
-            }
-        };
+        let builder = AppClientBuilder::new(grpc_client, app_config.clone());
 
-        let address: SocketAddr = "0.0.0.0:12346".parse().unwrap();
-        let tls = Box::new(Esp32Tls::new_server(&tls_server_config));
-        let tls_listener = Esp32Listener::new(address.into(), Some(tls)).unwrap();
+        let mut client = builder.build().unwrap();
 
-        let webrtc_certificate = Rc::new(webrtc_certificate);
-        let dtls = Esp32DtlsBuilder::new(webrtc_certificate.clone());
+        client.get_config().unwrap()
+    };
 
-        let cloned_exec = exec.clone();
+    let robot = match initializer {
```

Comment:
is it worth defining this as a function on the enum itself? So you could call `initializer.get_robot()`?

Reply:
hmm... yeah I see what you mean. worse-case it will make the code here more concise.

Reply:
actually, in the event of registry, you'll need a config response, which would need to be an optional parameter. Pulling this out to a separate function would need to be done on both entries and I'm not sure what would be the best way to reorganize things would be.

I think this approach is enough.

Reply:
ok